@xen-orchestra/web-core/lib/components/ui/progress-bar/UiProgressBar.vue

```diff
@@ -0,0 +1,114 @@
+<!-- v2 -->
+<template>
+  <div class="ui-progress-bar" :class="accentClass">
+    <div class="progress-bar">
+      <div class="fill" :style="{ width: `${fillWidth}%` }" />
+    </div>
+    <div v-if="shouldShowSteps" class="steps typo-body-regular-small">
+      <span>{{ $n(0, 'percent') }}</span>
+      <span v-for="step in steps" :key="step">{{ $n(step, 'percent') }}</span>
+    </div>
+    <VtsLegendList class="legend">
+      <UiLegend :accent :value="percentage" unit="%">{{ legend }}</UiLegend>
```

Comment:
When a value is used for computation, I suggest rounding it at the final step, for display only.

```suggestion
      <UiLegend :accent :value="Math.round(percentage)" unit="%">{{ legend }}</UiLegend>
```